homeassistant/components/fully_kiosk/entity.py

```diff
@@ -74,12 +74,14 @@ async def mqtt_subscribe(
         @callback
         def message_callback(message: mqtt.ReceiveMessage) -> None:
             payload = json.loads(message.payload)
-            event_callback(**payload)
+            if "event" in payload and payload["event"] == event:
```

Comment:
All the mqtt data and topic parsing really belongs in the 3rd party library. Please move it there in a follow up PR. Have the library accept callbacks for subscription (and publishing) but do all the details in the library.